### PR TITLE
[nop] Added documentation for some NASequence functions that needed it.

### PR DESCRIPTION
**Pull request by poshul**
## Description

Expanded documentation in NASequence.h

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/CHEMISTRY/NASequence.cpp [hunk 1]
```diff
@@ -267,12 +267,14 @@ namespace OpenMS
...
-    return getFormula(type, charge).getMonoWeight();
+    //getFormula adds Hydrogens, not protons, so we need to subtract the electrons
+    return getFormula(type, charge).getMonoWeight() - charge * Constants::ELECTRON_MASS_U;
   }
 
   double NASequence::getAverageWeight(NASFragmentType type, Int charge) const
   {
-    return getFormula(type, charge).getAverageWeight();
+    //getFormula adds Hydrogens, not protons, so we need to subtract the electrons
+    return getFormula(type, charge).getAverageWeight() - charge * Constants::ELECTRON_MASS_U;
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/NASequence.cpp:277]: What if charge is negative? Is still the correct mass calculated?

Comment (poshul) [reply src/openms/source/CHEMISTRY/NASequence.cpp:277]: Yes, in the negative charge case getFormula returns the formula of the uncharged species minus the hydrogens, so we need to add back the electron mass difference between the hydrogen and the proton, this is accomplished here by subtracting a negative number of ELECTRON_MASS_U

Comment (timosachsenberg) [reply src/openms/source/CHEMISTRY/NASequence.cpp:277]: Can you clarify this in the comment. Sounds like it adds always

**Comment by poshul** [comment]
I've tagged @hendrikweisser  on this since one of the places that GetMonoWeight is used with a charged nucleic acids in IdentificationDataConverter

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**